# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/Zone.h
# Commit: 410d1b57bb09
# Full Hash: 410d1b57bb0955e0fe2f47b95ffa5ee1005e6791
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-09 21:37:17
# Regressor Bug: 1643586
# File Overlap Count: 1
# Description:
#   Bug 1643586 - Part 5: Replace heap state checks that rely on there being a JSContext for the thread r=jandem
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D140192
# ==============================================================================

diff -r f78d139e1d0e -r 410d1b57bb09 js/src/gc/Zone.h
--- a/js/src/gc/Zone.h	Wed Mar 09 10:17:07 2022 +0000
+++ b/js/src/gc/Zone.h	Wed Mar 09 10:17:07 2022 +0000
@@ -367,12 +367,8 @@
     return isCollectingFromAnyThread();
   }
 
-  bool isCollectingFromAnyThread() const {
-    if (RuntimeHeapIsCollecting()) {
-      return wasGCStarted();
-    } else {
-      return needsIncrementalBarrier();
-    }
+  inline bool isCollectingFromAnyThread() const {
+    return needsIncrementalBarrier() || wasGCStarted();
   }
 
   bool shouldMarkInZone() const {